crates/model/src/order.rs

```diff
@@ -30,7 +30,7 @@ pub const BUY_ETH_ADDRESS: H160 = H160([0xee; 20]);
 /// An order that is returned when querying the orderbook.
 ///
 /// Contains extra fields that are populated by the orderbook.
-#[derive(Eq, PartialEq, Clone, Debug, Default, Deserialize, Serialize, Hash)]
+#[derive(Eq, PartialEq, Clone, Debug, Default, Deserialize, Serialize)]
```

Comment:
Not implementing `Hash` doesn't lead to problems (you only had to touch up a single piece of code) but isn't this an unusual way to handle that problem?
I think the common approach to hash an object with a unique identifier is to only consider that identifier for hashing. In that case all `Order`s with the same `OrderUid` would result in the same hash.
Coincidentally in the only existing usage of `Hash` this was the required behavior.

Reply:
So, my point was more that `Hash` doesn't really make sense for `Order` type because of a few reasons:

1. Data Integrity:
    If we wanted to have `Order: Hash` as you suggested use the `order_uid`, then I would say that you also probably need to have mechanisms (such as keeping fields private so they can't be modified) to protect from "invalid" orders from being created. Specifically, there is nothing preventing you from creating an order with a random `OrderUid` value, where it does not match the data at all. In that case, implementing `Hash` based on the `order_uid` parameter may lead to unexpected behaviour.
2. Clarity:
    There is some room for ambiguity in the `Hash` implementation. Imagine two identical orders, where the only difference is that one has slightly more executed than the other (the example in the PR description). They shouldn't `Hash` to the same value (my reasoning being that `Hash` should be an indicator of whether or not `a == b`), but it may make certain code unclear, where you assume that the same order (despite slightly different executions) will only be added once to the set. I would argue that for orders you should be using `HashMap<OrderUid, Order>` instead of `HashSet<Order>` as it forces you to decide how to deal with orders with the same UID but slightly different data.

So, I guess what I'm trying to say is that `Order: Hash` doesn't make that much semantic sense to me.

Reply:
I feel the problem is that there are different reasonable expectations on how Hash works with Orders. It could work like the smart contract or it could work like a Rust struct. Removing the Hash implementation fixes the problem. If you really need to Hash orders you can create your own wrapper struct that implements it in the way you prefer.